whotracksme/website/build/blog.py

```diff
@@ -22,59 +25,86 @@ def parse_blogpost(filepath):
-            template=get_template(data, "blog.html"),
-            blog_posts=[p for p in blog_posts if p['publish']]
-        ))
+    with open("_site/blog.html", "w") as output:
+        output.write(
+            render_template(
+                template=get_template(data, "blog.html"),
+                blog_posts=[p for p in blog_posts if p["publish"]],
+            )
+        )
     print_progress(text="Generate blog list")
 
 
 def build_blogpost_pages(data, blog_posts):
     for blog_post in blog_posts:
-        #TODO: Move template out after footnotes markdown extension does
+        # TODO: Move template out after footnotes markdown extension does
         # not save global state
         template = get_template(
-            data,
-            "blog-page.html",
-            render_markdown=True,
-            path_to_root='..'
+            data, "blog-page.html", render_markdown=True, path_to_root=".."
         )
-        with open(f'_site/blog/{blog_post.get("filename")}.html', 'w') as output:
+        with open(f'_site/blog/{blog_post.get("filename")}.html', "w") as output:
             output.write(
                 render_template(
-                    path_to_root='..',
-                    template=template,
-                    blog_post=blog_post
+                    path_to_root="..", template=template, blog_post=blog_post
                 )
             )
 
     print_progress(text="Generate blog posts")
 
 
 def get_human_date(date):
-    d = datetime.strptime(date, '%Y-%m-%d')
+    d = datetime.strptime(date, "%Y-%m-%d")
 
     if 4 <= d.day <= 20 or 24 <= d.day <= 30:
         suffix = "th"
     else:
         suffix = ["st", "nd", "rd"][d.day % 10 - 1]
     month = calendar.month_abbr[d.month]
     return f"{month} {d.day}{suffix}, {d.year}"
+
+
+def build_rss_feeds(blog_posts):
+    feed = FeedGenerator()
+    feed.load_extension("media", rss=True, atom=True)
+    base = "https://whotracks.me"
+
+    for post in blog_posts:
+        if post["publish"]:
+            entry = feed.add_entry()
+            entry.id(f'{base}/blog/{post["filename"]}.html')
+            entry.title(post["title"])
+            entry.link(link={"href": f"{base}/blog/{post['filename']}.html"})
+            entry.author({"name": post["author"]})
+            entry.pubDate(
+                datetime.strptime(post["date"], "%Y-%m-%d").replace(
+                    tzinfo=timezone("CET")
+                )
+            )
+            entry.description(post["subtitle"])
+            entry.media.thumbnail(
+                url=f'https://whotracks.me/static/img/{post["header_img"]}'
+            )
+
+    feed.title("WhoTracksMe blog")
+    feed.description("By the tech teams at Cliqz and Ghostery")
+    feed.link(link={"href": f"{base}/blog.html"})
+
+    feed.id("wtm")
+    feed.language("en")
+    feed.logo(f"{base}/static/img/who-tracksme-logo.png")
+
+    feed.rss_file("_site/blog/feed.rss.xml")
+    feed.atom_file("_site/blog/feed.atom.xml")
```

Comment:
Should these be accessible from the blog index.html somehow?

Reply:
Fair point - let's add them to the footer.